### ISSUE TITLE
Hardshrink and Softshrink use a strict comparison where ATen's zero band is closed, so the value and the gradient at exactly ±lambd are wrong

### ISSUE THREAD

**Issue by truong-v** [BobMcDear/attorch#15]
## Summary

For both shrink functions, ATen's zero band is the **closed** interval `[-lambd, lambd]`: at `|x| == lambd` the output is 0 and so is the gradient. attorch tests `|x| < lambd`, so at exactly the boundary `hardshrink` returns the input instead of 0, and both `hardshrink_grad` and `softshrink_grad` return a gradient of 1 instead of 0. `softshrink` is the clearest case, because the file disagrees with itself: the forward is written as a two-sided strict comparison and is flat on the closed band, so its derivative there is 0 — but its own gradient helper says 1.

`lambd` defaults to 0.5, and `±0.5` is an ordinary value in quantised or clipped activations, so this is not only a theoretical corner: `tests/test_act_layers.py` **fails on it today** at the larger shapes, where with ~24 million `randn` samples at least one lands exactly on `±lambd` and `assert_close` catches it. 88 of that file's `Hardshrink` and `Softshrink` parametrizations currently fail, and `tests/test_glu_layer.py` fails another 24 the same way, since these helpers are shared with `attorch.GLU`.

## Reproduction

```python
import torch
from torch import nn

import attorch

x = torch.tensor([-1.0, -0.5, -0.25, 0.0, 0.25, 0.5, 1.0], device='cuda')
row = lambda t: '  '.join(f'{v:8.4f}' for v in t.tolist())

for cls in ('Hardshrink', 'Softshrink'):
    print(cls)
    for name, module in (('torch  ', getattr(nn, cls)(0.5)),
                         ('attorch', getattr(attorch, cls)(0.5))):
        xx = x.clone().requires_grad_(True)
        y = module(xx)
        y.backward(torch.ones_like(y))
        print(f'  {name} forward ', row(y.detach()))
        print(f'  {name} gradient', row(xx.grad))
```

```
Hardshrink
  torch   forward   -1.0000    0.0000    0.0000    0.0000    0.0000    0.0000    1.0000
  torch   gradient   1.0000    0.0000    0.0000    0.0000    0.0000    0.0000    1.0000
  attorch forward   -1.0000   -0.5000    0.0000    0.0000    0.0000    0.5000    1.0000
  attorch gradient   1.0000    1.0000    0.0000    0.0000    0.0000    1.0000    1.0000
Softshrink
  torch   forward   -0.5000    0.0000    0.0000    0.0000    0.0000    0.0000    0.5000
  torch   gradient   1.0000    0.0000    0.0000    0.0000    0.0000    0.0000    1.0000
  attorch forward   -0.5000    0.0000    0.0000    0.0000    0.0000    0.0000    0.5000
  attorch gradient   1.0000    1.0000    0.0000    0.0000    0.0000    1.0000    1.0000
```

The second and sixth cells are `x = ∓lambd`; every interior point agrees, and 4096 random values agree exactly.

## Environment

- attorch at `88ed34d`
- NVIDIA L40S (sm_89)
- torch 2.4.0+cu121, triton 3.0.0, Python 3.10, driver CUDA 12.4
